### PR TITLE
Reduces number of fd open calls

### PR DESCRIPTION
**Pull request by EgorWeders**
Partial revert for 8bd554916a7ede43c3686dc52f0563424e1d6ff0 (???)

### CODE REVIEW

File: src/laser.c [hunk 1]
```diff
@@ -105,15 +105,20 @@ void laser_print_entry(struct laser_dirent *entry, const char *color,
...
-    else if (laser_checktype(filename, laser_mediaformats))
-        return LASER_COLOR_MEDIA;
-    else if (laser_checktype(filename, laser_documentformats))
-        return LASER_COLOR_DOCUMENT;
-
-    return LASER_COLOR_FILE;
+
+{ //create fd because we use lseek to return on Bof position each laser_checktype_magic call inside
+    // and we can keep fd while checking types, even if fd is not valid
+    int fd = open(filename, O_RDONLY);
```
Comment (jmattaa) [inline src/laser.c:111]: I don't see how this reduces the amount of fd open call, rather dosen't this increase the open calls? In the main branch we only call open if checking extension fails. This calls open before even checking on the extension.

File: src/filetypes/checktype.c [hunk 2]
```diff
@@ -13,18 +13,12 @@ enum check_type_return
     Unknown
 };
 
-int laser_checktype(const char *filename, const struct laser_filetype formats[])
+int laser_checktype_sniff(int fd, const char *filename,
```
Comment (jmattaa) [inline src/filetypes/checktype.c:16]: I'd feel that it's better to have `laser_checktype` function that first checks on the extension, if nothing is matched then we go for the file signatures.

**Review by jmattaa — COMMENTED** [review]
I don't really understand what this does except calling open for every entry. And not using the extensions at all.

**Comment by EgorWeders** [comment]
I've put this pr in a draft, no time to work on it yet.

I think better would be to make different paths for checks

1. Check all available formats extensions with filename
2. If 1 step fails open fd for sniffing
3. If 2 step succeeds try to check formats by content magic numbers on one (from 2 step we know fd status, and can lseek for every call)

**Pull request closed without merging**